api/src/utils/apply-query.ts

```diff
@@ -819,23 +820,35 @@ export function applyFilter(
 }
 
 export async function applySearch(
+	knex: Knex,
 	schema: SchemaOverview,
 	dbQuery: Knex.QueryBuilder,
 	searchQuery: string,
 	collection: string,
 ): Promise<void> {
+	const { number: numberHelper } = getHelpers(knex);
 	const fields = Object.entries(schema.collections[collection]!.fields);
 
 	dbQuery.andWhere(function () {
 		fields.forEach(([name, field]) => {
 			if (['text', 'string'].includes(field.type)) {
 				this.orWhereRaw(`LOWER(??) LIKE ?`, [`${collection}.${name}`, `%${searchQuery.toLowerCase()}%`]);
 			} else if (['bigInteger', 'integer', 'decimal', 'float'].includes(field.type)) {
-				const number = Number(searchQuery);
+				const number = parseNumericString(searchQuery);
+
+				if (number === null) {
+					return; // unable to parse
+				}
 
-				// only cast finite base10 numeric values
-				if (validateNumber(searchQuery, number)) {
-					this.orWhere({ [`${collection}.${name}`]: number });
+				// creating a new object for now because TS doesnt recuce the FieldOverview type
+				const numberOptions = {
+					type: field.type as NumericType,
+					precision: field.precision,
+					scale: field.scale,
+				};
+
+				if (numberHelper.isNumberValid(number, numberOptions)) {
```

Comment:
We have a couple approaches here, what do you think:
- this new object approach
- try and fix typescripts type reduction here with custom guards or smth
- simply casting the value
```suggestion
				if (numberHelper.isNumberValid(number, field as NumberInfo)) {
```

Reply:
I like (and implemented) the type guard route.